src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -17,54 +17,56 @@ import {
     PUSHPIN_STYLE_CIRCLE_SIZE,
     PUSHPIN_STYLE_CIRCLE_STROKE_COLOR,
     EMPTY_SEGMENT_VALUE,
+    PUSHPIN_SIZE_OPTIONS_MAP,
 } from "../../../constants/geoChart";
-import { IGeoData, IGeoPushPinConfig } from "../../../interfaces/GeoChart";
+import { IGeoData, IGeoPointsConfig, IGeoConfig } from "../../../interfaces/GeoChart";
 import { SEGMENT } from "../../../constants/bucketNames";
 import { getMinMax } from "../../../helpers/utils";
-import { pushpinSizeOptionsMap } from "./geoChartDataSource";
+import isEmpty = require("lodash/isEmpty");
 
 function getExpressionByBucketName(name: string): mapboxgl.Expression {
     return ["get", "value", ["object", ["get", name]]];
 }
 
 function createPushpinSizeOptions(
     geoData: IGeoData,
-    pushpinSizeOptions: IGeoPushPinConfig,
+    pushpinSizeOptions: IGeoPointsConfig,
 ): mapboxgl.Expression | number {
     const { size } = geoData;
     const hasSize = size !== undefined;
+    const defaultRadius = PUSHPIN_SIZE_OPTIONS_MAP.min.default / 2;
 
     if (!hasSize || size.data.length === 0) {
-        return pushpinSizeOptionsMap.min.default;
+        return defaultRadius;
     }
 
-    const { min: minSize, max: maxSize } = getMinMax(size.data);
-    if (maxSize === minSize) {
-        return pushpinSizeOptionsMap.min.default;
+    const { min: minSizeFromData, max: maxSizeFromData } = getMinMax(size.data);
+    if (maxSizeFromData === minSizeFromData) {
+        return defaultRadius;
     }
 
-    const { minSize: minSizeOption = "default", maxSize: maxSizeOption = "default" } =
+    const { minSize: minSizeFromConfig = "default", maxSize: maxSizeFromConfig = "default" } =
         pushpinSizeOptions || {};
-    const minSizeOptionInPixel = pushpinSizeOptionsMap.min[minSizeOption];
-    const maxSizeOptionInPixel = pushpinSizeOptionsMap.max[maxSizeOption];
-    const multiplier = Math.pow(maxSizeOptionInPixel / minSizeOptionInPixel, 0.25);
-    const getStopPointSize = partial(getPointSize, minSizeOptionInPixel, multiplier);
+    const minSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.min[minSizeFromConfig];
+    const maxSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.max[maxSizeFromConfig];
+    const base = Math.pow(maxSizeInPixel / minSizeInPixel, 0.25);
+    const getStopPointSize = partial(getPointSize, minSizeInPixel, base);
 
     // The mouseenter event uses queryRenderedFeatures to determine when the mouse is touching a feature
     // And queryRenderedFeatures is not supporting nested object in expression
     // https://github.com/mapbox/mapbox-gl-js/issues/7194
     return [
         "step",
-        ["get", "pushpinRadius"],
-        minSizeOptionInPixel, // a
+        ["get", "pushpinSize"],
```

Comment:
Please be careful to rename `pushpinRadius` to `pushpinSize` since this mapbox expression is refering to structure [here](https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/core/geoChart/geoChartDataSource.ts#L62)

If we insist renaming it, please do it both sides, or make a constant key for them

Reply:
i also changed all references, see [here](https://github.com/gooddata/gooddata-react-components/pull/1458/files#diff-6b220fe05c6963feb3422220c77bcfa9R75)

Reply:
By using `pushpinSize`, we don't know if this size is radius or diameter.
IMO, this change is not necessary :)